### DISCUSSION TITLE
Workflow

### DISCUSSION THREAD

**Discussion by ekhenderson** [cs481-ekh/f22-softskills#178] [Instructor Feedback]
Team,

Your main git branch is a mess :P

Please make sure you are following standard procedures when merging pull requests. You should be creating topic branches for your tasks through the github interface. When work is complete issue a pull requests. When pull requests are approved, use the github interface to **squash merge** them to the main branch, then delete the topic branch.

Also, make sure when you are merging the main branch into your topic branch that you are going the right way!

And finally, it looks like people are reviewing and approving their own pull requests ?! Not good practice!

Please make sure to bring this up at the next meeting so there is no confusion. I expect going forward that your main branch will be nice and linear. Thanks!

**Comment by TheBardUnit** [comment]
Just saw this post, will bring it up to group ASAP so we can address concerns.
